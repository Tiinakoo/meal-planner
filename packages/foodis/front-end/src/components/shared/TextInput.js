import React from "react";
import { observer } from "mobx-react-lite";

const TextInput = ({
  model = {},
  value = model.value,
  onChange = model.setValue,
  ...props
}) => {
  const onNativeChange = ({ target: { value } }) => onChange(value);

  return (
    <input
      className="input"
      value={value}
      onChange={onNativeChange}
      placeholder="Add new department"
      {...props}
    />
  );
};

export default observer(TextInput);
