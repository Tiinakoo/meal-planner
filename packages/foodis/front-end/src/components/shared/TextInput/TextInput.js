import React from "react";
import { observer } from "mobx-react-lite";
import styles from "./TextInput.module.scss";

const TextInput = ({
  model = {},
  value = model.value,
  onChange = model.setValue,
  ...props
}) => {
  const onNativeChange = ({ target: { value } }) => onChange(value);

  return (
    <input
      type="text"
      className={styles.input}
      value={value}
      onChange={onNativeChange}
      placeholder="Add new department"
      {...props}
    />
  );
};

export default observer(TextInput);
