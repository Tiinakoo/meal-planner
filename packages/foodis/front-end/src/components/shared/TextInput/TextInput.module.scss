@import '../../../styles/baseline';
@import '../../../styles/typography';

.input {
    padding: $baseline;
    margin: $baseline 0;
    border-radius: $baseline;
    font-size: 2 * $baseline;

    @include primaryFont;
}
