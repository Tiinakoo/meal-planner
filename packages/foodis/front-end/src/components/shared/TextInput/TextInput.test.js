import React from "react";

import { mount } from "enzyme";
import TextInput from "./TextInput";

describe("TextInput", () => {
  let component;
  let setValueMock;

  beforeEach(() => {
    setValueMock = jest.fn();

    const modelStub = {
      value: "some value",
      setValue: setValueMock,
    };

    component = mount(
      <TextInput
        label="Some label"
        placeholder="Some placeholder"
        model={modelStub}
        onChange={modelStub.setValue}
      />
    );
  });

  it("renders", () => {
    expect(component).toMatchSnapshot();
  });

  describe("an InputWrapper within", () => {
    let inputWrapper;

    beforeEach(() => {
      inputWrapper = component.find("InputWrapper");
    });

    it("knows label from a prop", () => {
      expect(inputWrapper).toHaveProp("label", "Some label");
    });
  });

  describe("an input within", () => {
    let input;

    beforeEach(() => {
      input = component.find("input");
    });

    it("knows value from model", () => {
      expect(input).toHaveProp("value", "some value");
    });

    it("knows placeholder from a prop", () => {
      expect(input).toHaveProp("placeholder", "Some placeholder");
    });

    it("when changed, abstracts event away and uses input value instead", () => {
      input.simulate("change", { target: { value: "some value" } });

      expect(setValueMock).toHaveBeenCalledWith("some value");
    });
  });
});
