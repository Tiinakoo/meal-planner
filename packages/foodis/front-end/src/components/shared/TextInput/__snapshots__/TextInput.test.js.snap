// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput renders 1`] = `
<TextInput
  label="Some label"
  model={
    Object {
      "setValue": [MockFunction],
      "value": "some value",
    }
  }
  onChange={[MockFunction]}
  placeholder="Some placeholder"
>
  <InputWrapper
    label="Some label"
  >
    <div>
      <div>
        <BodyText
          font="primary"
          size="md"
        >
          <p
            className=""
          >
            Some label
          </p>
        </BodyText>
      </div>
      <input
        onChange={[Function]}
        placeholder="Some placeholder"
        type="text"
        value="some value"
      />
    </div>
  </InputWrapper>
</TextInput>
`;
